systemtests/src/main/java/io/enmasse/systemtest/manager/ResourceManager.java

```diff
@@ -0,0 +1,441 @@
+    public static String ADDRESS_SPACE_TYPE;
+    public static String ADDRESS_SPACE_PLAN;
+
+    abstract void initFactories(AddressSpace addressSpace);
+
+    abstract void setup() throws Exception;
+
+    abstract void tearDown(ExtensionContext context) throws Exception;
+
+    public abstract AmqpClientFactory getAmqpClientFactory();
+
+    public abstract void setAmqpClientFactory(AmqpClientFactory amqpClientFactory);
+
+    public abstract MqttClientFactory getMqttClientFactory();
+
+    public abstract void setMqttClientFactory(MqttClientFactory mqttClientFactory);
+
+    public AddressSpace getSharedAddressSpace() {
+        return null;
+    }
+
+    //------------------------------------------------------------------------------------------------
+    // Client factories
+    //------------------------------------------------------------------------------------------------
+
+    public void closeClientFactories(AmqpClientFactory amqpClientFactory, MqttClientFactory mqttClientFactory) throws Exception {
+        if (amqpClientFactory != null) {
+            amqpClientFactory.close();
+        }
+        if (mqttClientFactory != null) {
+            mqttClientFactory.close();
+        }
+    }
+
+    //------------------------------------------------------------------------------------------------
+    // Address plans
+    //------------------------------------------------------------------------------------------------
+
+    public void createAddressPlan(AddressPlan addressPlan, boolean replaceExisting) throws Exception {
+        LOGGER.info("Address plan {} will be created {}", addressPlan.getMetadata().getName(), addressPlan);
+        var client = Kubernetes.getInstance().getAddressPlanClient();
+        if (replaceExisting) {
+            client.createOrReplace(addressPlan);
+        } else {
+            client.create(addressPlan);
+        }
+        Thread.sleep(1000);
+    }
+
+    public void removeAddressPlan(AddressPlan addressPlan) throws Exception {
+        Kubernetes.getInstance().getAddressPlanClient().withName(addressPlan.getMetadata().getName()).cascading(true).delete();
+    }
+
+    public void replaceAddressPlan(AddressPlan plan) throws Exception {
+        Kubernetes.getInstance().getAddressPlanClient().createOrReplace(plan);
+    }
+
+    public AddressPlan getAddressPlan(String name) throws Exception {
+        return Kubernetes.getInstance().getAddressPlanClient().withName(name).get();
+    }
+
+    //------------------------------------------------------------------------------------------------
+    // Address space plans
+    //------------------------------------------------------------------------------------------------
+
+    public void createAddressSpacePlan(AddressSpacePlan addressSpacePlan) throws Exception {
+        createAddressSpacePlan(addressSpacePlan, false);
+    }
+
+    public void createAddressSpacePlan(AddressSpacePlan addressSpacePlan, boolean replaceExisting) throws Exception {
```

Comment:
Suggested improvement for the future: how about having a replaceAddressSpacePlan() instead of the boolean parameter

Reply:
I didn't find any use for replacing address space plan. If we will have some test which will replace address space plan then i will for sure implement this method.